server/src/main/java/org/elasticsearch/search/dfs/AggregatedDfs.java

```diff
@@ -32,9 +34,17 @@ public AggregatedDfs(StreamInput in) throws IOException {
         termStatistics = HppcMaps.newMap(size);
         for (int i = 0; i < size; i++) {
             Term term = new Term(in.readString(), in.readBytesRef());
-            TermStatistics stats = new TermStatistics(in.readBytesRef(),
-                in.readVLong(),
-                DfsSearchResult.subOne(in.readVLong()));
+            BytesRef term2 = in.readBytesRef();
+            final long docFreq = in.readVLong();
+            assert docFreq >= 0;
+            long totalTermFreq = DfsSearchResult.subOne(in.readVLong());
```

Comment:
It'd be great to add a comment here explaining why this backwards compatibility logic is needed. I also wonder if it's worth factoring out a common method like `readSingleTermStats` to use in both `AggregatedDfs` and `DfsSearchResult`?

Reply:
> I also wonder if it's worth factoring out a common method like readSingleTermStats to use in both AggregatedDfs and DfsSearchResult

These methods are similar, but not quite the same. I'm therefore inclined to keep them separate.

> It'd be great to add a comment here explaining why this backwards compatibility logic is needed.

I have added docs to that effect in 0aa3cf78f08.